process-reports/src/test/scala/com/gu/contentapi/LambdaSpec.scala

```
@@ -0,0 +1,21 @@
+package com.gu.contentapi
+
+import com.gu.contentapi.services.S3
+import com.typesafe.scalalogging.LazyLogging
+import org.scalatest.FlatSpec
+
+class LambdaSpec extends FlatSpec with LazyLogging {
+
+  behavior of "run the lambda"
+
+  it should "test the app end to end" in {
+    val l = new Lambda
+
+    for {
+      reportNames <- S3.getLastTwoReportNames("uk")
+      lastReport <- S3.downloadReport(reportNames.lastOne)
+      secondLastReport <- S3.downloadReport(reportNames.secondLast)
```

For future thought: each of these functions returns Options, so if one was to fail for any reason it would fail silently at the moment.